### DISCUSSION TITLE
how to use pglite for e2e test ?

### DISCUSSION THREAD

**Discussion by elycheikhsmail** [electric-sql/pglite#228] [Q&A]
I'have project nextjs+ts that use postgres via query  called kyseyle, and cypress for e2e testing. I ask if possible to use pglite for e2e testing, run migration befor running test ?

**Comment by czeidler** [comment]
There is: https://www.npmjs.com/package/kysely-pglite 
You just have to configure it for your tests and run the migrations before doing your tests...

**Comment by samwillis** [comment]
Hey @elycheikhsmail 

PGlite works well for tests, either as you suggest creating an instance in a a before all, or you can even create an instance inside each test.

The PGlite test are a good place to look for inspiration: https://github.com/electric-sql/pglite/tree/main/packages/pglite/tests
